doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md

```diff
@@ -0,0 +1,455 @@
+
+Independently of the data generator, NGSI context data is always transformed into internal Flume events at cygnus-ngsi sources. In the end, the information within these Flume events must be mapped into specific CartoDB data structures at the Cygnus sinks.
+
+Next sections will explain this in detail.
+
+[Top](#top)
+
+###<a name="section1.1"></a>Mapping NGSI events to flume events
+Notified NGSI events (containing context data) are transformed into Flume events (such an event is a mix of certain headers and a byte-based body), independently of the NGSI data generator or the final backend where it is persisted.
+
+This is done at the cygnus-ngsi Http listeners (in Flume jergon, sources) thanks to [`NGSIRestHandler`](./orion_rest_handler.md). Once translated, the data (now, as a Flume event) is put into the internal channels for future consumption (see next section).
+
+[Top](#top)
+
+###<a name="section1.2"></a>Mapping Flume events to CartoDB data structures
+CartoDB is based on [PostgreSQL](http://www.postgresql.org/) and [PostGIS](http://postgis.net/) extensions. It organizes the data in databases (one per organization), schemas (una per project within an organization) and tables (a schema may have one or more tables). Such organization is exploited by `NGSICartoDBSink` each time a Flume event is going to be persisted.
+
+[Top](#top)
+
+####<a name="section1.2.1"></a>PostgreSQL databases and schemas naming conventions
+PostgreSQL databases and schemas are already created by CartoDB upon organization and project request, respectively. Thus, it is up to CartoDB to define the naming conventions for these elements.
+
+Here it is asseumed the notified/default FIWARE service maps the PostgreSQL schema/project name.
+
+[Top](#top)
+
+####<a name="section1.2.2"></a>PostgreSQL tables naming conventions
+The name of these tables depends on the configured data model (see the [Configuration](#section2.1) section for more details):
+
+* Data model by service path (`data_model=dm-by-service-path`). As the data model name denotes, the notified FIWARE service path (or the configured one as default in [`NGSIRestHandler`](.ngsi_rest_handler.md)) is used as the name of the table. This allows the data about all the NGSI entities belonging to the same service path is stored in this unique table. The only constraint regarding this data model is the FIWARE service path cannot be the root one (`/`).
+* Data model by entity (`data_model=dm-by-entity`). For each entity, the notified/default FIWARE service path is concatenated to the notified entity ID and entityType in order to compose the table name. The concatenation charated is `_` (underscore). If the FIWARE service path is the root one (`/`) then only the entity ID and type are concatenated.
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1022/commits/273a671ffa3dd8e412fccbf6abffd91742d05cdd